### ISSUE TITLE
Low accuracy

### ISSUE THREAD

**Issue by browarsoftware** [Luca96/face-clustering#2]
I have run github implementation of your method several times (I did full training 6 or 7 times, 12 epoches each) and I did not achive more than 80% of accurancy during training. What are "secret steps" I have to undertake to obtain 89% of accuracy in the first two epochs? :-) Are there some network tuning options that are not set on github?

**Comment by Luca96** [comment]
Hi, I've trained my [model](https://github.com/Luca96/face-clustering/blob/master/weights/weights-FC37-MobileNetV2-0.92.hdf5) with Google Colab's GPU for almost 8-10 epochs. I did different trials achieving almost the same accuracy (around 90%).

Anyway, You can try to:
1. **Debug** your loaded images: check the pixels' values, see if channels are switched. (are you training on CelebA, right?)
2. **Pretrain** on ImageNet:
```python
def mobilenet_model(num_features):
  base = MobileNetV2(input_shape=IMG_SHAPE,
                     weights='imagenet',     # <- 
                     include_top=False,
                     pooling='avg')
  
  # model top
  x = base.output
  x = Dense(1536, activation='relu')(x)
  x = BatchNormalization()(x)
  x = Dropout(0.3)(x)

  top = Dense(num_features, activation='sigmoid')(x)
  return Model(inputs=base.input, outputs=top)
```

3. Reduce the strength of **data augmentation** (or avoid it):
```python
# try with a factor of '0.1' instead of '0.2'
train_datagen = ImageDataGenerator(rotation_range=10, 
                                   rescale=1./255, 
                                   width_shift_range=0.1, 
                                   height_shift_range=0.1, 
                                   shear_range=0.1, 
                                   zoom_range=0.1, 
                                   horizontal_flip=True, 
                                   fill_mode='nearest')
```
4. If none of the above works, try a different optimizer (e.g. `adam`) and/or increase the `batch_size` to 96 or 128.

Hope it helps.

**Comment by browarsoftware** [comment]
Thank you for the response! :-)

(1) Yes
(2) I have changed weights='imagenet'
(3) I have removed data augmentation:

train_datagen = tf.keras.preprocessing.image.ImageDataGenerator(
                                   rescale=1. / 255,
    )

(4) Bigger batch than 64 does not fit into my video memory (as long as I correctly read the exception)

After 32 epoches I obtained 86%.
I run tensorflow 2.1 on GeForce 2600 GTX.

In model.compile I do not have 'cosine_proximity' - this option is unavilable. Instead I use 'cosine_similarity' - is this the same?

**Comment by Luca96** [comment]
Hi, I think that `cosine_proximity` should be the same as `cosine_similarity`. 

I've noticed you're using keras stuff from tensorflow (I mean `tf.keras`). So, that's why you cannot find `cosine_similarity` because it's not defined under the tf.keras namespace. But that's fine.

You can also try to use `binary_crossentropy` as loss function and see if accuracy improves.

> Note: if you import stuff from **tf.keras** make sure to use it for every keras-related stuff. Because there are some inconsistencies between plain keras imports and tf.keras ones. Also, I'm not aware of the existence of potential performance discrepancies between keras and tf.keras.

**Comment by browarsoftware** [comment]
Yes! Thank you! This was an issue - after installing keras everythng works fine. I had only to replace 'other' with 'raw', and change batch size to 32:

valid_generator = valid_datagen.flow_from_dataframe(
    dataframe=valid_split,
    directory=celeba.images_folder,
    x_col='image_id',
    y_col=celeba.features_name,
    target_size=TARGET_SIZE,
    batch_size=batch_size,
    #class_mode='other'
    class_mode='raw'
)

It seems that tf.keras and keras works differently.
